packages/scoutgame/src/builderNfts/recordNftMint.ts

```diff
@@ -213,6 +214,83 @@ export async function recordNftMint(
     log.error('Error completing quest', { error, builderId: builderNft.builderId, questType: 'scout-starter-card' });
   }
 
+  try {
+    // Find the referrer of the scout first
+    const referrer = await prisma.scout.findFirst({
+      where: {
+        events: {
+          some: {
+            referralCodeEvent: {
+              refereeId: scoutId
+            }
+          }
+        }
+      },
+      select: {
+        id: true
+      }
+    });
+
+    if (referrer) {
+      const referralCodeEvent = await prisma.referralCodeEvent.findFirst({
```

Comment:
We do'nt have a database guarantee that refereeId is unique; you should only do one query. i dont think u need to query referrer first. just query the referallcodeevent